File: Knjigoteka.Model/Responses/ReviewResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Knjigoteka.Model.Responses
{
    public class ReviewResponse
    {
        public int Id { get; set; }
        public int BookId { get; set; }
        public string BookTitle { get; set; } = null!;
        public int UserId { get; set; }
        public string UserFullName { get; set; } = null!;
        public int Rating { get; set; }
        public string? Comment { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}

File: Knjigoteka.UI/knjigoteka_mobile/lib/models/review_response.dart
class ReviewResponse {
  final int id;
  final int bookId;
  final String bookTitle;
  final int userId;
  final String userFullName;
  final int rating;
  final DateTime createdAt;

  ReviewResponse({
    required this.id,
    required this.bookId,
    required this.bookTitle,
    required this.userId,
    required this.userFullName,
    required this.rating,
    required this.createdAt,
  });

  factory ReviewResponse.fromJson(Map<String, dynamic> json) => ReviewResponse(
    id: json['id'],
    bookId: json['bookId'],
    bookTitle: json['bookTitle'],
    userId: json['userId'],
    userFullName: json['userFullName'],
    rating: json['rating'],
    createdAt: DateTime.parse(json['createdAt']),
  );
}
